pkg/api/vlabs/types.go

```diff
@@ -519,6 +519,7 @@ type CustomCloudProfile struct {
 	AzureEnvironmentSpecConfig *AzureEnvironmentSpecConfig `json:"azureEnvironmentSpecConfig,omitempty"`
 	IdentitySystem             string                      `json:"identitySystem,omitempty"`
 	AuthenticationMethod       string                      `json:"authenticationMethod,omitempty"`
+	DependenciesLocation       string                      `json:"dependenciesLocation,omitempty"`
```

Comment:
Let's make DependenciesLocation a type since we only accept specific values (see Distro and OSType as examples)